tests/arch/arm/arm_thread_swap/src/arm_thread_arch.c

```diff
@@ -68,25 +68,31 @@ static void load_callee_saved_regs(const _callee_saved_t *regs)
 	__asm__ volatile (
 		"mov r1, r7;\n\t"
 		"mov r0, %0;\n\t"
+		"add r0, #16;\n\t"
 		"ldmia r0!, {r4-r7};\n\t"
+		"mov r8, r4;\n\t"
+		"mov r9, r5;\n\t"
+		"mov r10, r6;\n\t"
+		"mov r11, r7;\n\t"
+		"sub r0, #32;\n\t"
 		"ldmia r0!, {r4-r7};\n\t"
 		"mov r8, r4;\n\t"
 		"mov r9, r5;\n\t"
 		"mov r10, r6;\n\t"
 		"mov r11, r7;\n\t"
 		"mov r7, r1;\n\t"
-		:
+		: /* no output */
 		: "r" (regs)
-		: "memory"
+		: "memory", "r1", "r0", "r8", "r9", "r10", "r11"
```

Comment:
the purpose of this function is to load the content of `regs` into `v1-v8` (i.e. `r4-r11`), so does it really make sense to put these into the clobber list?

Reply:
We only need r1 and r0 here, you're right.

Reply:
This first case seems clear, it should only have r0 + r1, which I've updated. The discussion is around the much more complicated second case.